src/main/java/frc/team2412/robot/subsystem/IndexSubsystem.java

```diff
@@ -1,11 +1,190 @@
+        this.feederMotor.configFactoryDefault();
+
+        this.ingestMotor.setNeutralMode(NeutralMode.Coast);
+        this.feederMotor.setNeutralMode(NeutralMode.Coast);
+
+        this.ingestMotor.configSupplyCurrentLimit(MAX_MOTOR_CURRENT);
+        this.feederMotor.configSupplyCurrentLimit(MAX_MOTOR_CURRENT);
+
+        ingestMotorStop();
+        feederMotorStop();
+
+        ingestBallState = HAS_NO_BALL;
+        feederBallState = HAS_NO_BALL;
+
+        setName("IndexSubsystem");
+    }
+
+    // Methods
+
+    /**
+     * Spins first motor inward and updates first motor state
+     */
+    public void ingestMotorIn() {
+        ingestMotor.set(INDEX_IN_SPEED);
+        ingestMotorState = IN;
+    }
+
+    /**
+     * Spins first motor outward and updates first motor state
+     */
+    public void ingestMotorOut() {
+        ingestMotor.set(INDEX_OUT_SPEED);
+        ingestMotorState = OUT;
+    }
+
+    /**
+     * Stops first motor and updates first motor state
+     */
+    public void ingestMotorStop() {
+        if (!ingestSensorHasBallIn()) {
+            ingestMotor.set(0);
+            ingestMotorState = STOPPED;
+        }
+    }
+
+    /**
+     * Spins second motor inward and updates second motor state
+     */
+    public void feederMotorIn() {
+        feederMotor.set(INDEX_IN_SPEED);
+        feederMotorState = IN;
+    }
+
+    /**
+     * Spins second motor outward and updates second motor state
+     */
+    public void feederMotorOut() {
+        feederMotor.set(INDEX_OUT_SPEED);
+        feederMotorState = OUT;
+    }
+
+    /**
+     * Stops second motor and updates second motor state
+     */
+    public void feederMotorStop() {
+        feederMotor.set(0);
+        feederMotorState = STOPPED;
+    }
+
+    /**
+     * Checks if ball is positioned at the first sensor
+     */
+    public boolean ingestSensorHasBallIn() { // also might rename later?
+        return (ingestColorSensor.getProximity() > PROXIMITY_THRESHOLD); // value not 1500, to be determined actual
+                                                                            // value
+    }
+
+    /**
+     * Checks if ball is positioned at the second sensor
+     */
+    public boolean feederSensorHasBallIn() { // might rename methods later?
+        return (feederColorSensor.getProximity() > PROXIMITY_THRESHOLD); // value not 1500, to be determined actual
+                                                                            // value
+    }
+
+    public boolean isIngestMotorOn() {
+        return !(ingestMotorState == STOPPED);
+    }
+
+    public boolean isFeederMotorOn() {
+        return !(feederMotorState == STOPPED);
+    }
+
+    // do need now! :D D: :3 8) B) :P C: xD :p :] E: :} :>
+    @Override
+    public void periodic() {
+        if (ingestSensorHasBallIn()) {
+            ingestBallState = HAS_BALL;
+        } else {
+            ingestBallState = HAS_NO_BALL;
+        }
+
+        if (feederSensorHasBallIn()) {
```

Comment:
No need to have conditionals.  Should be able to set ingestBallState to ingestSensorHasBallIn();